cmd/snap-discard-ns/snap-discard-ns.c

```diff
@@ -41,22 +42,61 @@
 #endif
 
 int main(int argc, char** argv) {
-    if (argc != 2) {
-        printf("Usage: snap-discard-ns <SNAP-INSTANCE-NAME>\n");
-        return 0;
+    bool from_snap_confine = false;
+    const char* snap_instance_name = NULL;
+    int optind;
+
+    for (optind = 1; optind < argc; ++optind) {
+        const char* opt = argv[optind];
+        if (sc_streq(opt, "--from-snap-confine")) {
+            from_snap_confine = true;
+            continue;
+        }
+        if (sc_streq(opt, "--")) {
+            /* Start parsing positional arguments. */
+            optind++;
+            break;
+        }
+        if (strlen(opt) > 0 && opt[0] == '-') {
+            errno = 0;
+            die("unsupported option: %s", opt);
+        }
+        break;
+    }
+    for (/* continue */; optind < argc; ++optind) {
+        const char* opt = argv[optind];
+        if (strlen(opt) > 0 && opt[0] == '-') {
```

Comment:
The whole point of `--` was to indicate that the rest is positionals, even if those start with `-` and could be misinterpreted as an option. Eg. `snap-discard-ns -- -foo` with `-foo` as snap instance name, which is invalid anyway and will be caught by `snap_instance_name_validate()`.

Feels like a simple if as proposed before would be enough.

Reply:
Oh, and there's is a nice and simple suggestion from @mvo5 too.